src/transformState.js

```diff
@@ -5,7 +5,26 @@
  * @param {Object[]} actions
  */
 function transformState(state, actions) {
-  // write code here
-}
+  for (const action of actions) {
+    if (action.type === 'addProperties') {
+      const extraData = action.extraData;
+
+      for (const key in extraData) {
+        state[key] = extraData[key];
+      }
+    } else if (action.type === 'removeProperties') {
+      const keysToRemove = action.keysToRemove;
 
+      for (const key of keysToRemove) {
```

Comment:
redundant variable 
please check other places 
```suggestion
      for (const key of action.keysToRemove) {
```